frontend/src/pages/login.tsx

```diff
@@ -1,19 +1,69 @@
+import { signInWithEmailAndPassword } from "firebase/auth";
+import { AlertCircle } from "lucide-react";
 import Image from "next/image";
-import { ReactElement } from "react";
+import { useRouter } from "next/navigation";
+import { ReactElement, useState } from "react";
 import { FieldValues, SubmitHandler, useForm } from "react-hook-form";
 
+import { GET } from "@/api/requests";
 import Landing from "@/components/Landing";
 import { Textfield } from "@/components/Textfield";
 import { Button } from "@/components/ui/button";
+import { initFirebase } from "@/firebase/firebase";
 import { useWindowSize } from "@/hooks/useWindowSize";
 import { cn } from "@/lib/utils";
 
 export default function Login() {
-  const { register, setValue, handleSubmit } = useForm();
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
   const _setValue = setValue;
 
+  const [firebaseError, setFirebaseError] = useState("");
+
+  const { auth } = initFirebase();
+  const router = useRouter();
+
+  const login = async (email: string, password: string) => {
+    return await signInWithEmailAndPassword(auth, email, password)
+      .then((userCredential) => {
+        return userCredential.user.getIdToken();
+      })
+      .then((token) => {
+        return token;
+      });
+  };
+
+  const sendTokenToBackend = async (token: string) => {
+    try {
+      const response = await GET(`http://localhost:4000/api/user/`, {
```

Comment:
Might want to change the url to not be localhost

Reply:
Do you want me to put the backend URL as a env variable?

Reply:
If you look at this [line](https://github.com/TritonSE/PIA-Program-Manager/blob/main/frontend/src/api/requests.ts#L10) on main we already have the different base URL configurations setup. I think you should be able to call the route by doing `GET('user/', ...)`. You can look at how the [student page ](https://github.com/TritonSE/PIA-Program-Manager/blob/main/frontend/src/api/students.ts#L47) works for reference.

Reply:
Realized that there was an error in main because the base url was incorrect. Fixed it tho

Reply:
I think it was supposed to be an env variable. Not sure if it got deleted or never modified on the google drive.

Reply:
Yeah that would be good. I tried modifying the URL but when I did it looks like this function throws some error. Essentially after the logging in the home page student table doesn't load. However, if I were to access the home page without logging in then the student table does load, so I think there's some bug with this function and not the student table.